runtime/commonMain/src/kotlinx/serialization/protobuf/ProtoBuf.kt

```diff
@@ -40,6 +40,9 @@ class ProtoBuf(context: SerialModule = EmptyModule) : AbstractSerialFormat(conte
             else -> throw SerializationException("Primitives are not supported at top-level")
         }
 
+        override fun shouldWriteNullElement(desc: SerialDescriptor, tag: ProtoDesc, index: Int): Boolean =
+            desc.kind !is StructureKind.CLASS
+
```

Comment:
This is a very minimal change in `ProtoBuf` but perhaps the format should have more changes (and `Tagged` none). The correct interception point is in `encodeNullableSerializableValue` that has full flexibility on how to encode the nullable value.
Basically I see 4 main approaches for handling null values in a format like ProtoBuf that doesn't support them (the choice should be configurable):

- Disallow them (current solution, safe but not always satisfactory)
- Use a marker flag (most correct solution, but there is no standard - other options should be available when the format is not under full control) -- write an "isNull" flag and only write the value if the flag has a false value, the NullableSerializer class is designed in this philosophy
- Omit null values - this has as disadvantages that no distinction can be made between missing values (which need to be set to the defaults) and null values (which need to be set to `null`)
- Use "null" objects (I don't think this works for protobuf) - where there is an actual object that gets the meaning of null (this means that object itself cannot be serialized)